exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -115,8 +117,12 @@ final void write(Collection<MetricData> metrics, OutputStream output) throws IOE
                             PrometheusType.forMetric(metric)),
                     LinkedHashMap::new,
                     Collectors.toList()));
+    Optional<Resource> optResource = metrics.stream().findFirst().map(MetricData::getResource);
```

Comment:
Just check isempty and set to null when it is possibly with ternary. Don't need a stream to do that ;)